src/log-pcap.c

```diff
@@ -1077,6 +1292,103 @@ static OutputInitResult PcapLogInitCtx(ConfNode *conf)
             }
             SCLogInfo("Using log dir %s", pl->dir);
         }
+
+        const char *compression_str = NULL;
+        compression_str = ConfNodeLookupChildValue(conf, "compression");
+        if (compression_str == NULL) {
+            compression_str = DEFAULT_PCAP_LOG_COMPRESSION_FORMAT;
+        }
+
+        PcapLogCompressionData *comp = &pl->compression;
+        if (strcmp(compression_str, "none") == 0) {
```

Comment:
if (compression_str == NULL || strcmp(compression_str, "none") == 0) {

would make lines 1298 to 1300 unnecessary